### PR TITLE
Added Sub lists for both ordered and unordered

### PR DESCRIPTION
**Pull request by itechdivyanshu**
#43 #29 
- Commit
    - [91f1c6d](https://github.com/editor-js/list/commit/91f1c6da8ce297157f72e5576a2d82a29b55af72)

- Feature Preview

ezgif com-video-to-gif

### CODE REVIEW

**Comment by ItechIndrustries** [comment]
Hey @PolinaShneider @neSpecc @talyguryn Some one please review this request

File: src/index.js [hunk 1]
```diff
@@ -404,6 +439,33 @@ class List {
...
+    }
+
+    const style = this._data.style === 'ordered' ? this.CSS.wrapperOrdered : this.CSS.wrapperUnordered;
+    var ol = this._make('ul', [this.CSS.baseBlock, this.CSS.wrapper, style], {
+      contentEditable: true,
+    });
+    if (this.currentItem.nextSibling != null) {
+      this.currentItem.parentNode.insertBefore(ol, this.currentItem.nextSibling) 
+    }
+    else{
```
Comment (talyguryn) [inline src/index.js:460]: Please, check code-style: spaces, blanks lines and line brakes.

**Review by talyguryn — CHANGES_REQUESTED** [review]
You should not try to change package name from `@editorjs/list` to `@itech-indrustries/list` for this repo. 

Please, check code-style: spaces, blanks lines and line brakes. Change all `var` to `let` or `const`.

File: src/index.js [hunk 2]
```diff
@@ -302,16 +307,26 @@ class List {
...
-
-    for (let i = 0; i < items.length; i++) {
-      const value = items[i].innerHTML.replace('<br>', ' ').trim();
-
-      if (value) {
-        this._data.items.push(items[i].innerHTML);
+    const itemsList = this._elements.wrapper.childNodes;
+
+    function getData(items){
+      let dataEach = [];
```
Comment (talyguryn) [inline src/index.js:313]:
```suggestion
      let dataEach = [];
      
```

File: src/index.js [hunk 3]
```diff
@@ -302,16 +307,26 @@ class List {
...
+      for (let i = 0; i < items.length; i++) {
+        const value = items[i].innerHTML.replace('<br>', ' ').trim();
+  
+        if (items[i].tagName == "UL") {
+          dataEach.push(getData(items[i].childNodes));
+        }
```
Comment (talyguryn) [inline src/index.js:319]:
```
} else if (value) {
```

File: src/index.js [hunk 4]
```diff
@@ -339,6 +354,26 @@ class List {
...
 
+
+  createAllElm(lidata){
+    const style = this._data.style === 'ordered' ? this.CSS.wrapperOrdered : this.CSS.wrapperUnordered;
+    const ulElem = this._make('ul', [this.CSS.baseBlock, this.CSS.wrapper, style], {
+      contentEditable: true,
+    });
+
+    lidata.forEach((item) => {
+
```
Comment (talyguryn) [inline src/index.js:365]:
```suggestion
```

File: src/index.js [hunk 5]
```diff
@@ -339,6 +354,26 @@ class List {
...
+      if(typeof(item) == "object"){
```
Comment (talyguryn) [inline src/index.js:366]:
```suggestion
      if (typeof(item) === "object") {
```

File: src/index.js [hunk 6]
```diff
@@ -339,6 +354,26 @@ class List {
...
+        ulElem.appendChild(this.createAllElm(item))
+      }else{
```
Comment (talyguryn) [inline src/index.js:368]:
```suggestion
      } else {
```

File: src/index.js [hunk 7]
```diff
@@ -339,6 +354,26 @@ class List {
...
+        ulElem.appendChild(this._make('li', this.CSS.item, {
+          innerHTML: item,
+        }));
+      } 
+    });
+    return ulElem;
```
Comment (talyguryn) [inline src/index.js:374]:
```suggestion

    return ulElem;
```

File: src/index.js [hunk 8]
```diff
@@ -404,6 +439,34 @@ class List {
     }
   }
 
+  /**
+   * Indent the List Items
+   *
+   * @param {KeyboardEvent} event
+   */
+  addTab(event){
+
```
Comment (talyguryn) [inline src/index.js:448]:
```suggestion
```

File: src/index.js [hunk 9]
```diff
@@ -404,6 +439,34 @@ class List {
...
+    if (this.currentItem == this.currentItem.parentNode.childNodes[0]) {
```
Comment (talyguryn) [inline src/index.js:449]:
```suggestion
    if (this.currentItem === this.currentItem.parentNode.childNodes[0]) {
```

**Review by talyguryn — CHANGES_REQUESTED** [review]

**Comment by talyguryn** [comment]
after pressing enter at the end of sublist there is a problem with margins for a new items in the main list

image

**Comment by Froxz** [comment]
#16

**Comment by Froxz** [comment]
@talyguryn @PolinaShneider any chance this gets merged or requires more fixes?

**Comment by neSpecc** [comment]
Resolved by making an advanced [Nested List](https://github.com/editor-js/nested-list) tool

**Pull request closed without merging**